web/components/answers/answer-resolve-panel.tsx

```diff
@@ -117,36 +120,73 @@ export function AnswerResolvePanel(props: {
-                : `${answers.length} answers`
-            }
-            marketTitle={contract.question}
-            disabled={
-              !resolveOption ||
-              (resolveOption === 'CHOOSE' && !answers.length) ||
-              (resolveOption === 'CHOOSE_MULTIPLE' &&
-                (!(answers.length > 1) ||
-                  !answers.every((answer) => chosenAnswers[answer] > 0)))
-            }
-            onResolve={onResolve}
-            isSubmitting={isSubmitting}
-          />
+          {!modalSetOpen && (
+            <ResolveConfirmationButton
+              color={
+                resolveOption === 'CANCEL'
+                  ? 'yellow'
+                  : resolveOption === 'CHOOSE' && answers.length
+                  ? 'green'
+                  : resolveOption === 'CHOOSE_MULTIPLE' &&
+                    answers.length > 1 &&
+                    answers.every((answer) => chosenAnswers[answer] > 0)
+                  ? 'blue'
+                  : 'indigo'
+              }
+              label={
+                resolveOption === 'CANCEL'
+                  ? 'N/A'
+                  : resolveOption === 'CHOOSE'
+                  ? chosenText
+                  : `${answers.length} answers`
+              }
+              marketTitle={contract.question}
+              disabled={
+                !resolveOption ||
+                (resolveOption === 'CHOOSE' && !answers.length) ||
+                (resolveOption === 'CHOOSE_MULTIPLE' &&
+                  (!(answers.length > 1) ||
+                    !answers.every((answer) => chosenAnswers[answer] > 0)))
+              }
+              onResolve={onResolve}
+              isSubmitting={isSubmitting}
+            />
+          )}
+          {modalSetOpen && (
+            <Button
+              color={
+                resolveOption === 'CANCEL'
+                  ? 'yellow'
+                  : resolveOption === 'CHOOSE' && answers.length
+                  ? 'green'
+                  : resolveOption === 'CHOOSE_MULTIPLE' &&
+                    answers.length > 1 &&
+                    answers.every((answer) => chosenAnswers[answer] > 0)
+                  ? 'blue'
+                  : 'indigo'
+              }
+              disabled={
+                isSubmitting ||
+                !resolveOption ||
+                (resolveOption === 'CHOOSE' && !answers.length) ||
+                (resolveOption === 'CHOOSE_MULTIPLE' &&
+                  (!(answers.length > 1) ||
+                    !answers.every((answer) => chosenAnswers[answer] > 0)))
+              }
+              onClick={onResolve}
+            >
+              <>
+                Resolve{' '}
+                {resolveOption === 'CANCEL' ? (
+                  <>N/A</>
+                ) : resolveOption === 'CHOOSE' ? (
+                  chosenText
+                ) : (
+                  <>{answers.length} answers</>
+                )}
+              </>
+            </Button>
```

Comment:
This duplication of code / logic is not great. At the least, you should create variables for color, disabled, and button label defined above that you reuse.